### PR TITLE
🌱 Set UserAgent field in all controller managers

### PR DESCRIPTION
**Pull request by ykakarap**
**What this PR does / why we need it**:
Sets the user agent for the managers.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

This is a follow up of https://github.com/kubernetes-sigs/cluster-api/pull/4257
Closes #4257

This PR cherry-picks the original authors commit and adds a commit to address the review comments on the PR.

### CODE REVIEW

**Comment by enxebre** [comment]
Supersedes https://github.com/kubernetes-sigs/cluster-api/pull/4257/files

File: test/infrastructure/docker/main.go [hunk 1]
```diff
@@ -102,7 +103,9 @@ func main() {
 
 	ctrl.SetLogger(klogr.New())
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	restConfig := ctrl.GetConfigOrDie()
+	restConfig.UserAgent = remote.DefaultClusterAPIUserAgent("cluster-api-docker-controller")
```
Comment (enxebre) [inline test/infrastructure/docker/main.go:107]: `cluster-api-docker-controller` consistently with all the others?

**Comment by vincepri** [comment]
/retitle 🌱 Set UserAgent field in all controller managers

File: test/infrastructure/docker/main.go (hunk 1 above)
Comment (ykakarap) [reply test/infrastructure/docker/main.go]: 👍🏼. Addressed.

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**